base/loading.jl

```diff
@@ -3131,7 +3131,7 @@ function replace_depot_path(path::AbstractString)
             depot = dirname(depot)
         end
 
-        if startswith(path, depot)
+        if startswith(path, string(depot, Filesystem.pathsep())) || path == depot
```

Comment:
In many cases, Windows accepts `/` as a path separator - is there any risk of that appearing here?

Reply:
oh, I did not know that.
I pushed a fix with an accompanying test.

Reply:
Thanks! The replace isn't safe for UNC paths (it destroys double slashes `\\`), but maybe we don't support those?

https://learn.microsoft.com/en-us/dotnet/standard/io/file-path-formats#example-ways-to-refer-to-the-same-file shows some exotic ways to build paths on Windows, in case we do.

Reply:
I now settled to just use `abspath` to normalize both.
That should also increase the accuracy in matching depots in cases where one is an absolute and the other a relative one.

Reply:
I traced back the use of `replace_depot_path` and found that these issues should already be taken care of.
In particular,
- the `path` argument will be absolute and normalized when it arrives, which happens at `base/loading.jl:_include_dependency`,
- the elements of `DEPOT_PATH` also went through `abspath` before they are handed over to precompilation, see `base/loading.jl:create_expr_cache`.

So I will revert changes related to this and make the tests pass again.

Reply:
Where is `DEPOT_PATH` normalized if I do `--output-ji` with `JULIA_DEPOT_PATH`?

It seems like that would also hit this code path, iiuc

Reply:
Right before spawning the precompilation process, here https://github.com/JuliaLang/julia/blob/f2f76d8833541091ef13a47c49bb77b2263d9937/base/loading.jl#L2812

Shouldn't `--output-ji` hit that too?

Reply:
That's something that spawns a process with `--output-ji` not the other way around

Reply:
I am sorry, I don't understand.
Where would be the entry point for what you describe?

Reply:
Ah, my point is that it's legal to spawn Julia with `--output-ji` outside of the spawning that precompilation does, and in that case the DEPOT_PATH may not be normalized

Reply:
Ok, I see. Indeed, that's problematic.